File: 1683-maximum-number-of-coins-you-can-get/maximum-number-of-coins-you-can-get.py
class Solution:
    def maxCoins(self, piles: List[int]) -> int:
        i = 0
        j = len(piles)-1
        piles.sort()
        me = 0
        while i < j:
            bob = piles[i]
            alice = piles[j]
            j-=1
            me += piles[j]
            i+=1
            j-=1
        return me

        

File: 1683-maximum-number-of-coins-you-can-get/maximum-number-of-coins-you-can-get.java
class Solution {
    public int maxCoins(int[] piles) {
        int i = 0, j = piles.length-1;
        int me = 0;
        Arrays.sort(piles);
        while(i<j) {
            int bob = piles[i++];
            int alice = piles[j--];
            me += piles[j--];
        }
        return me;
    }
}